### PR TITLE
Credentials cannot be unrevoked with StatusList2021

### PR DESCRIPTION
**Pull request by UMR1352**
# Description of change
Fixed a "bug" that allows for revoked credentials to become valid again.

## Type of change
- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## How the change has been tested
Added a unit test.

## Change checklist
- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by eike-hass — APPROVED** [review]

File: identity_credential/src/revocation/status_list_2021/credential.rs [hunk 1]
```diff
@@ -34,9 +34,12 @@ pub enum StatusList2021CredentialError {
   /// Inner status list failures.
   #[error(transparent)]
   StatusListError(#[from] StatusListError),
-  /// Missing status list id
+  /// Missing status list id.
   #[error("Cannot set the status of a credential without a \"credentialSubject.id\".")]
   Unreferenceable,
+  /// Credentials cannot be unrevoked.
+  #[error("A previously revoked credential cannot be unrevoked.")]
+  UnrevocableCredential,
```
Comment (abdulmth) [inline identity_credential/src/revocation/status_list_2021/credential.rs:42]:
```suggestion
  UnreversibleRevocation,
```

File: identity_credential/src/revocation/status_list_2021/credential.rs [hunk 2]
```diff
@@ -403,4 +415,33 @@ mod tests {
...
+
+    assert!(status_list_credential.set_entry(420, false).is_ok());
+    status_list_credential.set_entry(420, true).unwrap();
+    assert_eq!(
+      status_list_credential.set_entry(420, false),
+      Err(StatusList2021CredentialError::UnrevocableCredential)
+    );
+  }
+  #[test]
+  fn suspended_credential_can_be_unsuspended() {
```
Comment (abdulmth) [inline identity_credential/src/revocation/status_list_2021/credential.rs:436]: You're not suspending the credential here first.

**Review by abdulmth — CHANGES_REQUESTED** [review]

File: identity_credential/src/revocation/status_list_2021/credential.rs (hunk 1 above)
Comment (UMR1352) [reply identity_credential/src/revocation/status_list_2021/credential.rs:42]: Like it

File: identity_credential/src/revocation/status_list_2021/credential.rs [hunk 3]
```diff
@@ -117,6 +120,11 @@ impl StatusList2021Credential {
 
   /// Sets the credential status of a given [`Credential`],
   /// mapping it to the `index`-th entry of this [`StatusList2021Credential`].
+  ///
+  /// ## Note:
+  /// - A revoked credential cannot ever be unrevoked and will lead to a
+  /// [`StatusList2021CredentialError::UnrevocableCredential`].
+  /// - Trying to unrevoke an already valid credential won't do anything.
```
Comment (abdulmth) [inline identity_credential/src/revocation/status_list_2021/credential.rs:127]:
```suggestion
  /// - Trying to set `revoked_or_suspended` to `false` for an already valid credential will have no impact.
```

File: identity_credential/src/revocation/status_list_2021/credential.rs [hunk 4]
```diff
@@ -117,6 +120,11 @@ impl StatusList2021Credential {
 
   /// Sets the credential status of a given [`Credential`],
   /// mapping it to the `index`-th entry of this [`StatusList2021Credential`].
+  ///
+  /// ## Note:
+  /// - A revoked credential cannot ever be unrevoked and will lead to a
+  /// [`StatusList2021CredentialError::UnrevocableCredential`].
```
Comment (abdulmth) [inline identity_credential/src/revocation/status_list_2021/credential.rs:126]:
```suggestion
  /// [`StatusList2021CredentialError::UnreversibleRevocation`].
```

**Review by abdulmth — APPROVED** [review]

**Pull request merged**